app/components/explore/side-bar.js

```diff
@@ -16,9 +26,57 @@ export default Component.extend({
     },
 
     dateValidate(date) {
-      if (moment(date).isAfter(this.get('endDate'))) {
-        this.set('endDate', date);
+      if (moment(date).isAfter(this.get('customEndDate'))) {
+        this.set('customEndDate', date);
       }
+
+      this.send('selectDateFilter', 'custom_dates');
+    },
+
+    selectDateFilter(dateType) {
+      let newStartDate = null;
+      let newEndDate = null;
+
+      if (dateType === this.get('dateType') && dateType !== 'custom_dates') {
```

Comment:
Personally I think using a switch statement here would make this more readable. We may go with this if others are ok with this.

Reply:
I would switch to polymorphism rather than a switch statement.

But if this is just going to be used once, it's fine as it is, no need to complicate it more.

Reply:
Please use the switch statements. They are meant to be used for long chain of `if/else if` only.